app/src/main/java/ch/epfl/polybazaar/UI/SubmitOffer.java

```diff
@@ -0,0 +1,99 @@
+package ch.epfl.polybazaar.UI;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.google.firebase.Timestamp;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import ch.epfl.polybazaar.R;
+import ch.epfl.polybazaar.chat.ChatMessage;
+import ch.epfl.polybazaar.listing.Listing;
+import ch.epfl.polybazaar.litelisting.LiteListing;
+import ch.epfl.polybazaar.login.AuthenticatorFactory;
+
+import static ch.epfl.polybazaar.chat.ChatMessage.OFFER_ACCEPTED;
+import static ch.epfl.polybazaar.chat.ChatMessage.OFFER_PROCESSED;
+import static ch.epfl.polybazaar.listing.Listing.SOLD;
+import static java.util.UUID.randomUUID;
+
+public class SubmitOffer extends AppCompatActivity {
+
+    public static final String LISTING = "LISTING";
+    private EditText inputOffer;
+    private Listing listing;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_submit_offer);
+        inputOffer = findViewById(R.id.offer);
+        listing = (Listing)getIntent().getSerializableExtra(LISTING);
+    }
+
+    public void submitOffer(View view) {
+        if (!String.valueOf(inputOffer.getText()).isEmpty()) {
+            double offer = Double.parseDouble(inputOffer.getText().toString());
+            sendOffer(offer, listing, SubmitOffer.this);
+            finish();
+        } else {
+            Toast.makeText(getApplicationContext(), R.string.offer_invalid, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public void cancelOfferMaking(View view) {
+        finish();
+    }
+
+    /**
+     * sends an offer
+     * @param offer the offer amount
+     */
+    public static void sendOffer(Double offer, Listing listing, Activity activity) {
+        String senderEmail = AuthenticatorFactory.getDependency().getCurrentUser().getEmail();
+        ChatMessage message = new ChatMessage(senderEmail, listing.getUserEmail(),
+                listing.getId(),
+                ChatMessage.OFFER_MADE + offer.toString(),
+                new Timestamp(new Date(System.currentTimeMillis())));

```

Comment:
Not sure which format you need but there is the function Timestamp.now()

Reply:
I used this to stay coherent with the other sent messages. But I am sure that Timestamp.now() would work just as well.